#!/usr/bin/env python
PACKAGE = "bobi_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("num_agents",    int_t,    0, "Number of individuals to track", 1,  0, 100)
gen.add("num_robots",    int_t,    0, "Number of robots to track", 1,  0, 5)
gen.add("num_background_samples",    int_t,    0, "Number of background samples for the subtractor", 100,  0, 1000)
gen.add("bstor_history",    int_t,    0, "History of frames for the subtractor", 500,  0, 1000)
gen.add("var_threshold",    double_t,    0, "Variance threshold", 4,  0, 1000)
gen.add("detect_shadows",   bool_t,   0, "Detect shadows",  False)
gen.add("learning_rate",    double_t,    0, "Learning rate for the subtractor", 0.05,  0, 1)
gen.add("min_contour_size",    int_t,    0, "Minimum size of contours to accept", 10,  0, 1000)

gen.add("quality_level",    double_t,    0, "Quality level for good features to track", 0.01,  0, 1000)
gen.add("min_distance",    double_t,    0, "Minimume distance between corners to accept", 5,  0, 500)
gen.add("block_size",    int_t,    0, "Minimnum block size for good features to track", 5,  0, 100)
gen.add("use_harris_dtor",   bool_t,   0, "Use Harris detector",  True)
gen.add("k",    double_t,    0, "Harris k value", 0.04,  0, 2)
gen.add("reset_background_dtor",   bool_t,   0, "Reset background detector",  False)

gen.add("threshold_new_value",    int_t,    0, "Value for binary threshold", 80,  0, 255)
gen.add("threhold_value",    int_t,    0, "Value for pixels under the threshold", 150,  0, 255)

exit(gen.generate(PACKAGE, "individual_tracking_node", "BlobDetector"))